### ISSUE TITLE
Icons instead of letters on VBL/MBL drawing tristate button

### ISSUE THREAD

**Issue by irisiflimsi** [RPTools/maptool#2284]
**Is your feature request related to a problem? Please describe.**
The VBL / MBL drawing tristate button uses letters, where symbols are readily available. I.e. the eye and a boot.

**Describe the solution you'd like**
Use an eye instead of V and a boot instead of M. This should fit into the 25x25 button.

**Describe alternatives you've considered**
Stay with the buttons as is. But the letters are not easier to interpret than the icons mentioned above.

**Additional context**
Consider the SVG support feature request. Using that would mean writing a lot of new icons, though.

**Comment by irisiflimsi** [comment]
Proposals:
![mbl-only](https://user-images.githubusercontent.com/2022972/97899031-063e0000-1d39-11eb-91f6-965691f93784.png)
![mbl-vbl-on](https://user-images.githubusercontent.com/2022972/97899040-09d18700-1d39-11eb-81a7-8891d301a5ef.png)
![vbl-only](https://user-images.githubusercontent.com/2022972/97899049-0c33e100-1d39-11eb-8049-7a3663494565.png)

**Comment by Azhrei** [comment]
As discussed in another issue somewhere, the two shades of blue are a bad choice for people with colorblindness, so I'd like to see those changed.  Otherwise, I'd be happier to see separate buttons for VBL and MBL as a single tristate button is just too cluttered.  IMO, of course.

**Comment by irisiflimsi** [comment]
That's true for almost all the buttons. The buttons have too little contrast and waste screen space by being confined to a circle. (Screenspace is also the reason why more modern GUIs no longer have the 3d l&f to them.) These eyesores have become a characteristic for maptool. :-) If I create more appropriate buttons, they don't blend in. And if you turn the tristate into two bistates, you have to deal with the no-draw-at-all state. At this point I only want to fix one problem at a time.

**Comment by JamzTheMan** [comment]
It would be great to replace all the icons really with something cleaner.

I had the same issue when i needed to update the template buttons, i could not represent enough data into those 25x25 round buttons so went with larger square buttons. 

I dont think anyone has issues with an icon revamp, we just need the graphic art to be done...

**Comment by Azhrei** [comment]
Agreed.  If you want to fix the icon for this issue, that's great, but we need an artist to create new (scalable?) buttons.  Then the user can select the size that suits them.  (Some UIs allow for icon titles to be displayed under the icon, but since we have tooltips for every one, I'm not sure it's quite so important.)

The nice thing about an SVG version of the icons is that they'd be super theme-able by editing theme elements into the SVG before rendering...

**Comment by irisiflimsi** [comment]
Then I'll be creating a pull request with these icons. Do we have artists looking for (free) work? That would go into a more general request, possibly being realized after #2283 . BTW, how can you detect that an issue has an approved pull request, if you don't close it?

**Comment by irisiflimsi** [comment]
Maybe some older preliminary version are available as SVG? This problem may have come up before but then SVG wasn't in the core? We could try and salvage something. Maybe from the time the logo was designed? Although with gradients and 3D-ish, we could work from there, if anything was saved. Is there?

**Comment by Azhrei** [comment]
Not that I know of.  I have a collection of logos (now part of a repo, I believe) but they're all PNG, nothing in SVG.

**Comment by irisiflimsi** [comment]
Maybe ask on discord for an SVG button redo? Maybe even the forum? The needed images are easy to collect, Sort of like the i18n properties. Don't you wish there was something like crowdin for icons?

**Comment by Azhrei** [comment]
Oh boy, that would be a mess!  Can you imagine a few dozen people each contributing a few icons in SVG and then someone else trying to organize them into an icon set that match each other?

Probably a much better route would be to have a single person do the entire set, or maybe one person coordinating an overall look and feel and then parceling out groups of icons to individual artists...

**Comment by irisiflimsi** [comment]
I don't have a feeling for the size of the community. A few dozen all eager to contribute is probably too many. 3-4 would be a good group. If a single person were willing to do the whole lot, that's probably okay too, Do you expect overwhelming response? My experience is that when the time comes to get real, there will not be that many left.

**Comment by bubblobill** [comment]
There are buttons everywhere, on the main screen and in various panes.
Please provide a wishlist of buttons and their functions to avoid ending up with gaps in the provided art.

**Comment by rtakehara** [comment]
I am willing to contribute, I even started doing icon sets  (in SVG, but at the time I exported them to raster images because that's the format used today) I had to stop for personal reasons but I could go back to it right now

My aproach to it was doing 1 color icons, it would avoid low contrast icons, and provide simpler, more flexible designs I could share them here again

**Comment by rtakehara** [comment]
[MT icons.zip](https://github.com/RPTools/maptool/files/5530852/MT.icons.zip)

**Comment by irisiflimsi** [comment]
The current development branch already supports SVG, they will be rendered in their "native" size. That means the user units are pixels. Images are in the src/main/resources directory and then mostly in image subdirectories. A handful are JPG, another handful are GIF but the overwhelming mass are PNG. I don't know, whether all are eligible for a redo. The icons of the initial discussion are all in src/main/resources/net/rptools/maptool/client/image/tool/.

That's the wishlist. I think. rtakehara graphically answered the other questions from bubblobill. :-) 

I like rtakehara's icons better than the remixicons, maybe help fill in the gaps there?

One artistic comment. Maybe we can pick up a color or a gradient from the main "supericon". Even though its not 1 color and may reduce the contrast slightly, it will preserve some reminiscence of the old 3d l&f and hint at the "corporate identity". It may be almost imperceptible at the 25x25 size and shouldn't be stressed, but it will "feel" like it. What do you think?

All my personal thoughts. What say you, project leaders?

**Comment by Azhrei** [comment]
> [...] they will be rendered in their "native" size. That means the user units are pixels.

There is code in MapTool that unpacks the JPG/PNG/GIF images into a `BufferedImage` and I presume that's where the conversion from vector to raster would be performed.  For reference, this page has sample code for converting an SVG into a JPEG:  https://xmlgraphics.apache.org/batik/using/transcoder.html

See https://jwatt.org/svg/authoring/ for ideas on authoring SVG (note that most tools will take these into account, but it seems Inkscape doesn't follow them).

> I don't know, whether all are eligible for a redo.

AFAIAC, everything is on the table. 🙂

> One artistic comment. Maybe we can pick up a color or a gradient from the main "supericon". Even though its not 1 color and may reduce the contrast slightly, it will preserve some reminiscence of the old 3d l&f and hint at the "corporate identity". It may be almost imperceptible at the 25x25 size and shouldn't be stressed, but it will "feel" like it. What do you think?

I'm not sure what you mean by "its not 1 color"?

I opened Rod's zipped SVG in Inkscape.  Looks like he covered pretty much every icon (although I didn't count them or match them up against the repo's contents).

Inkscape used some approaches that the authoring document discourages.  It may be that the authoring document is out of date or it could simply be that it's easier inside Inkscape to generate the SVG that way...

Since the transcoder allows hints to control the rasterization process, such as `ImageTranscoder.KEY_USER_STYLESHEET_URI`, we should be able to specify the color(s) to the used (for example, with `svg { background-color: ...; color: ...; }`).  This would let MapTool store theme colors into a stylesheet for use in conversion..

**Comment by bubblobill** [comment]
Inkscape is fully featured but annoying, especially in its default settings. Files should be saved as something other than the default. I know that saving as Optimised SVG allows using XML properties instead of CSS style code.

**Comment by bubblobill** [comment]
[examples.zip](https://github.com/RPTools/maptool/files/5535648/examples.zip)
Whilst most of the icons in any OS collection are going to be pretty naff, there are usually some good ones.
I modified the "eye" and the "guideline" in this lot but the other two are stock and may be suitable.

**Comment by irisiflimsi** [comment]
> There is code in MapTool that unpacks the JPG/PNG/GIF images into a `BufferedImage` and I presume that's where the conversion from vector to raster would be performed. For reference, this page has sample code for converting an SVG into a JPEG: https://xmlgraphics.apache.org/batik/using/transcoder.html

Unpacking is done through a java internal service register. Batik registers itself through dependency as being able to unpack svg. That renders SVG at "native" size, If you want something more fancy, code needs to be rewritten.

> > One artistic comment. Maybe we can pick up a color or a gradient from the main "supericon". Even though its not 1 color and may reduce the contrast slightly, it will preserve some reminiscence of the old 3d l&f and hint at the "corporate identity". It may be almost imperceptible at the 25x25 size and shouldn't be stressed, but it will "feel" like it. What do you think?
> I'm not sure what you mean by "its not 1 color"?

That's from Rod, and his icon set was drawn that way.

> I opened Rod's zipped SVG in Inkscape. Looks like he covered pretty much every icon (although I didn't count them or match them up against the repo's contents).

I have the feelingthere are 2-3 times as many icons in the source than in his SVG. But maybe I have double vision after looking down one too many code subdirectories.

**Comment by rtakehara** [comment]
> I have the feelingthere are 2-3 times as many icons in the source than in his SVG. But maybe I have double vision after looking down one too many code subdirectories.

yeah, I covered probably about half, not all, but I think some of those images are not used anymore, and new were added, so yeah, needs more icons for sure

**Comment by irisiflimsi** [comment]
Some clarification: I think maptool-dock-icon.png should be reused - that is what I meant with "corporate identity". It has a green gradient under the sextant. This gradient i meant  being "picked up" by the icon frames that are currently 1-color. I failed to create a simple example with masks in inkscape, unfortunately. Hopefully my description suffices. It may even be something that can be themed with enough svg-fu.

**Comment by JamzTheMan** [comment]
FYI we have twelve monkey lib in our code base which means all of these formats are generally accessable via normal io image reader unless you need to access specifics of a format.

List of options (but read the caveat at the top):  https://gist.github.com/dmnsgn/76878ba6903cf15789b712464875cfdc

**Comment by irisiflimsi** [comment]
Decoders must be added individually, they are not part of the core. (That's what the pull request contains. And the added test case failed without the inclusion.)

**Comment by kwvanderlinde** [comment]
With all the main icons being updated in 1.13, I think even the expanded discussion in thie FR has been addressed. Was there anything that was missed that would warrant keeping this open?
